src/html.js

```diff
@@ -80,14 +80,24 @@ export function HTMLString(str) {
 	this.value = str;
 }
 
-// adapted from TiddlyWiki <http://tiddlywiki.com> and Python 3's `html` module
+const HTML_ENCODE_REPLACE = {
+	"&": "&amp;",
+	"<": "&lt;",
+	">": "&gt;",
+	"\"": "&quot;",
+	"'": "&#x27;"
+};
+
 export function htmlEncode(str, attribute) {
-	let res = str.replace(/&/g, "&amp;").
-		replace(/</g, "&lt;").
-		replace(/>/g, "&gt;");
-	if(attribute) {
-		res = res.replace(/"/g, "&quot;").
-			replace(/'/g, "&#x27;");
+	const rx = attribute ? /[&<>'"]/g : /[&<>]/g;
+	let res;
+	// fast path for strings which don't need to be encoded
+	if(rx.test(str)) {
```

Comment:
Is it really worth running `#test` in addition to `#replace`? It seems to me that might be slower overall than just using `#replace` all the time:

```javascript
let res = str.replace(rx, c => HTML_ENCODE_REPLACE[c]);
```

A [cursory check](https://jsben.ch/D8DXB) suggests that `#replace` is almost as fast as `#test` when there's no match.

Reply:
I think you're probably right, maybe I did a little too much magic. Let me run some benchmarks without fastpath.

Reply:
Thanks to your intention i have done more benchmarking and research, ... with unexpected results ;)
On Firefox our "without fastpath" approach seems to be the best (most simple & good speed). But when it comes to chrome, things get different.

### On Chrome
When the content needs to be encoded, the current complate implementation outperform every approach we discussed so far, specially for non-attributes where the second branch will not be executed. But it's the slowest when it comes to attributes which don't have to be encoded, which should be in most cases. 

Just click this [Benchmark](https://jsben.ch/1sfpZ) on Chrome. That would bench the best possible input & engine for the current implementation. To make it easier to try out different inputs, i have put the parameters on top of the setup block. 

### New approach (loop based)
I have done more research and tried a loop based approach from [html-entities](https://github.com/mdevils/html-entities) (MIT License) and adopted to our needs. This implemtation tends to be the fastest under various inputs & engines.

Reply:
I will provide a new commit, but not today. Time to sleep^^